src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts

```diff
@@ -1273,6 +1278,58 @@ declare namespace Xrm {
         isOnPremises(): boolean;
     }
 
+    /**
+     * Interface for the result of calling Utility.getPageContext from an entity record
+     */
+    interface EntityRecordPageContext {
+        input: {
+            /**
+             * The current page type. The value returned is entityrecord or entitylist
+             */
+            pageType: "entityrecord" | "entitylist";
```

Comment:
The value of `pageType` is what can be used to tell between EntityRecordPageContext and EntityListPageContext.
So, the type can be hard-coded to either value, and can then be used in a type assersion.

```suggestion
            pageType: "entityrecord";
```

Reply:
Of course! Done for both